src/pocketmine/permission/Permission.php

```diff
@@ -92,16 +92,24 @@ public static function loadPermissions(array $data, string $default = self::DEFA
 	}
 
 	/**
-	 * @param string $name
-	 * @param array  $data
-	 * @param string $default
-	 * @param array  $output
-	 *
+	 * @param string      $name
+	 * @param array       $data
+	 * @param string      $default
+	 * @param array       $output
+	 * @param null|string $parentName
 	 * @return Permission
 	 *
 	 * @throws \Exception
 	 */
-	public static function loadPermission(string $name, array $data, string $default = self::DEFAULT_OP, array &$output = []) : Permission{
+	public static function loadPermission(string $name, array $data, string $default = self::DEFAULT_OP, array &$output = [], ?string $parentName = null) : Permission{
+		if($name{0} === "."){
+			if($parentName === null){
+				throw new \InvalidStateException("Permission names must not start with a dot");
```

Comment:
Not sure this is an invalid state condition. Perhaps invalid argument? It's bad data...

Reply:
Idk, blame @shoghicp. All the other `throw`s in this function throw `InvalidStateException`. I do the same just for consistency.

Reply:
there's literally one other throw... that makes no sense. "Let's be consistently wrong because blame @shoghicp"